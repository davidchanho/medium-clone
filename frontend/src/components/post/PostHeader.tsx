import React from "react";
import { Card } from "react-bootstrap";
import { useFindPublicationByPostId } from "../../hooks/useFindPublicationbyPostId";
import { PostProps } from "../../types";

function PostHeader({ post, className }: PostProps) {
  const publication = useFindPublicationByPostId(post.publicationId);

  return (
    <Card.Text className={`d-flex align-items-end ${className}`}>
      <img src={publication?.icon} className="mr-1 rounded" /> name in{" "}
      {publication?.name}
    </Card.Text>
  );
}

export default PostHeader;
