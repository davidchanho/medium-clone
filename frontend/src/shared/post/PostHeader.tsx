import React from "react";
import { Card } from "react-bootstrap";
import { useFindPublicationByPostId } from "../../hooks/useFindPublicationbyPostId";
import { PostProps } from "../../store/posts/types";

function PostHeader({ post, className }: PostProps) {
  const publication = useFindPublicationByPostId(post.publicationId);

  if(!post) {
    return null
  }

  return (
    <Card.Text className={`d-flex align-items-end mb-1 ${className}`}>
      <img src={publication?.icon} className="mr-1 rounded" /> name in{" "}
      {publication?.name}
    </Card.Text>
  );
}

export default PostHeader;
