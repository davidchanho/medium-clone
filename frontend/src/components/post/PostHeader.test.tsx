import React from "react";
import renderer from "react-test-renderer";
import { testPost } from "../../tests";
import PostHeader from "./PostHeader";

describe("PostHeader", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<PostHeader post={testPost}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
