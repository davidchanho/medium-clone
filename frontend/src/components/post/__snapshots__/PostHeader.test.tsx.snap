// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeader renders correctly 1`] = `
<p
  className="d-flex align-items-end mb-1 undefined card-text"
>
  <img
    className="mr-1 rounded"
    src="test"
  />
   
  test
   in 
  test
</p>
`;
